tests/futures.rs

```diff
@@ -0,0 +1,94 @@
+#[cfg(feature = "futures-traits")]
```

Comment:
Instead of the mod here, I think you can just make this `#![cfg(...)]` (note the `#!` instead of just `#`).